plugins/module_utils/nae.py

```diff
@@ -1109,13 +1108,16 @@ def new_compliance_requirement(self):
             self.result['Result'] = final_msg
 
     def new_compliance_requirement_set(self):
-        ag = self.get_assurance_group(self.params.get('ag_name'))["uuid"]
+        ag = self.get_assurance_group(self.params.get('ag_name'))
+        if ag is None:
+            self.result['Result'] = "No such Assurance Group exists"
+            self.module.fail_json(msg='Assurance group {0} does not exist'.format(self.params.get('ag_name')), **self.result)
+        self.params['fabric_uuid'] = str(ag.get('uuid'))
```

Comment:
will it break the logic if we use "str(ag.get('uuid'))" instead of "self.getFirstAG().get("uuid")"?

Reply:
Correct but the existing logic was bad. We need to query the right AG.